Source/Core/VideoCommon/AbstractTexture.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "Common/CommonTypes.h"
+#include "Common/MathUtil.h"
+#include "VideoCommon/TextureConfig.h"
+#include "VideoCommon/VideoCommon.h"
+
+class AbstractTexture
+{
+public:
+  const TextureConfig config;
+
+  explicit AbstractTexture(const TextureConfig& c)
```

Comment:
The definitions for the constructor and destructor should be in a cpp file. The destructor in particular can be defaulted in said cpp file.